src/IO/WriteBufferFromS3MemoryStream.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AWS_S3
+
+#include <iostream>
+
+namespace DB
+{
+
+struct MemoryStream: std::iostream
```

Comment:
Let's rename `MemoryStream` -> `StdStreamFromMemory` ? Because the whole idea of it is to make a `std::iostream`-compatible wrapper around a block in memory.